src/components/DataSession/DataSession.vue

```diff
@@ -85,11 +85,6 @@ function deleteOperation(operationID) {
 
 async function addOperation(operationDefinition) {
   const url = dataSessionsUrl + props.data.id + '/operations/'
-  if ('input_files' in operationDefinition.input_data) {
-    for (const image of operationDefinition.input_data.input_files) {
-      image.source = 'archive'
```

Comment:
Can't remember so: does `image.source` have an assigned value prior to this?

Reply:
yeah its gets that property when its sent back from the archive or datalab, however this code was overwriting the image.source = 'datalab' with 'archive' which was wrong

Reply:
gotcha! then LGTM 🎉

Reply:
we may need to still add a method for patching in missing sources but this code was arbitrarily replacing all image sources with 'archive' which isn't what we want to happen